packages/miniflare/src/plugins/shared/index.ts

```diff
@@ -191,10 +193,15 @@ export function getPersistPath(
 	// keep Miniflare 2's behaviour, so persist to a temporary path which we
 	// destroy on `dispose()`.
 	const memoryishPath = path.join(tmpPath, pluginName);
-	if (persist === undefined || persist === false) {
+	if (persist === false) {
 		return memoryishPath;
 	}
 
+	// If `persist` is undefined, use either the default path or fallback to the tmpPath
+	if (persist === undefined) {
+		return path.join(defaultPersistRoot ?? tmpPath, pluginName);
```

Comment:
This would be easier for me to follow if it was:

```suggestion
		return defaultPersistRoot === undefined ? memoryishPath : path.join(defaultPersistRoot, pluginName);
```